pkgs/top-level/octave-packages.nix

```diff
@@ -0,0 +1,92 @@
+# This file contains the GNU Octave add-on packages set.
+# Each attribute is an Octave library.
+# Expressions for the Octave libraries are supposed to be in `pkgs/development/octave-modules/<name>/default.nix`.
+
+{ pkgs
+, lib
+, stdenv
+, fetchurl
+, newScope
+, octave
+, lapack, blas
+, gfortran
+, autoreconfHook
+, python3
+, python3Packages
+, jdk
+, gnuplot
+, texinfo
+, nettle
+}:
+
+let
+
+  isOctaveFull = octave.enableQt;
+  isOctaveJIT = octave.enableJIT;
+
+  buildOctaveLibrary = callPackage ../development/interpreters/octave/mk-octave-derivation.nix {
+    inherit pkgs lib stdenv;
+    namePrefix = "${octave.pname}-${octave.version}";
+    inherit octave;
+  };
+
+  callPackage = pkgs.newScope {
+    inherit (pkgs) lib stdenv;
+    inherit buildOctaveLibrary;
+    inherit fetchurl;
+    inherit gnuplot texinfo;
+  };
+
+in rec {
+
+  control = callPackage ../development/octave-modules/control {
+    gfortran = gfortran;
+    autoreconfHook = autoreconfHook;
+    lapack = lapack;
+    blas = blas;
+  };
+
+  io = callPackage ../development/octave-modules/io {
+    enableJava = true;
+    jdk = jdk;
+  };
+
+  level-set = callPackage ../development/octave-modules/level-set { };
+
+  linear-algebra = callPackage ../development/octave-modules/linear-algebra { };
+
+  ltfat = callPackage ../development/octave-modules/ltfat {
+    fftw = pkgs.fftw;
+    fftwSinglePrec = pkgs.fftwSinglePrec;
+    fftwFloat = pkgs.fftwFloat;
+    fftwLongDouble = pkgs.fftwLongDouble;
+    lapack = lapack;
+    blas = blas;
+    portaudio = pkgs.portaudio;
+    jre = pkgs.jre;
+  };
+
+
+  signal = callPackage ../development/octave-modules/signal {
+    control = control;
+  };
+
+  symbolic = callPackage ../development/octave-modules/symbolic {
+    # Need to use sympy 1.5.1 for https://github.com/cbm755/octsympy/issues/1023
+    # It has been addressed, but not merged yet.
+    pythonEnv = (let
+      overridenPython = let
+        packageOverrides = self: super: {
+          sympy = super.sympy.overridePythonAttrs (old: rec {
+            version = pkgs.python2Packages.sympy.version;
+            src = pkgs.python2Packages.sympy.src;
+          });
+        };
+      in python3.override {inherit packageOverrides; self = overridenPython; };
+    in overridenPython.withPackages (ps: [
+      ps.sympy
+      ps.mpmath
```

Comment:
Can you compose this inside the expression? Again, ideally with `octave.python` which should be available in the passthru of it..

Reply:
When using `nix repl`, I don't see any mention of `octave.python` or `octave.passthru.python`. Did you meant that I define it myself?

Reply:
Sorry. Doing that rebase, and just figured out what you meant by `octave.passthru.python`. Working on it.

However, I'm not sure what you mean by "compose this inside the expression". Can you explain further?

Reply:
> However, I'm not sure what you mean by "compose this inside the expression". Can you explain further?

I meant that you can define `pythonEnv` inside the expression of this package, it's just another dependency.

Reply:
See https://github.com/NixOS/nixpkgs/pull/108562/commits/276ff6e14e6cf7a56bd723cdd03cf596b6c9f476